contracts/src/v0.8/functions/tests/1_0_0/FunctionsRouter.t.sol

```diff
@@ -1,111 +1,354 @@
-    address[] memory proposedContractSetAddresses = new address[](2);
-    proposedContractSetAddresses[0] = address(s_functionsCoordinator);
-    proposedContractSetAddresses[1] = address(s_termsOfServiceAllowList);
-
-    s_functionsRouter.proposeContractsUpdate(proposedContractSetIds, proposedContractSetAddresses);
-    s_functionsRouter.updateContracts();
+    s_functionsRouter.pause();
+  }
+
+  event Unpaused(address account);
+
+  function test_Unpause_RevertIfNotOwner() public {
+    // Send as stranger
+    vm.stopPrank();
+    vm.startPrank(STRANGER_ADDRESS);
+
+    vm.expectRevert("Only callable by owner");
+    s_functionsRouter.unpause();
   }
+
+  function test_Unpause_Success() public {
+    vm.expectEmit(false, false, false, true);
+    emit Unpaused(OWNER_ADDRESS);
+
+    s_functionsRouter.unpause();
+
+    bool isPaused = s_functionsRouter.paused();
+    assertEq(isPaused, false);
+  }
+}
+
+// ================================================================
+// |                    Functions Subscriptions                   |
+// ================================================================
+
+/// @notice #constructor
+contract FunctionsSubscriptions_Constructor {
+
 }
 
-contract FunctionsRouter_createSubscription is FunctionsSetRoutes {
+/// @notice #_markRequestInFlight
+contract FunctionsSubscriptions__MarkRequestInFlight {
+
+}
+
+/// @notice #_pay
+contract FunctionsSubscriptions__Pay {
+
+}
+
+/// @notice #ownerCancelSubscription
+contract FunctionsSubscriptions_OwnerCancelSubscription {
+
+}
+
+/// @notice #recoverFunds
+contract FunctionsSubscriptions_RecoverFunds {
+
+}
+
+/// @notice #oracleWithdraw
+contract FunctionsSubscriptions_OracleWithdraw {
+
+}
+
+/// @notice #ownerWithdraw
+contract FunctionsSubscriptions_OwnerWithdraw {
+
+}
+
+/// @notice #onTokenTransfer
+contract FunctionsSubscriptions_OnTokenTransfer {
+
+}
+
+/// @notice #getTotalBalance
+contract FunctionsSubscriptions_GetTotalBalance {
+
+}
+
+/// @notice #getSubscriptionCount
+contract FunctionsSubscriptions_GetSubscriptionCount {
+
+}
+
+/// @notice #getSubscription
+contract FunctionsSubscriptions_GetSubscription {
+
+}
+
+/// @notice #getConsumer
+contract FunctionsSubscriptions_GetConsumer {
+
+}
+
+/// @notice #_isExistingSubscription
+contract FunctionsSubscriptions__IsExistingSubscription {
+
+}
+
+/// @notice #_isAllowedConsumer
+contract FunctionsSubscriptions__IsAllowedConsumer {
+
+}
+
+/// @notice #createSubscription
+contract FunctionsSubscriptions_createSubscription is FunctionsOwnerAcceptTermsOfService {
   function setUp() public virtual override {
-    FunctionsSetRoutes.setUp();
+    FunctionsOwnerAcceptTermsOfService.setUp();
   }
 
   event SubscriptionCreated(uint64 indexed subscriptionId, address owner);
 
-  function testCreateSubscriptionSuccess() public {
-    vm.expectEmit();
-    emit SubscriptionCreated(1, OWNER);
+  function test_CreateSubscription_Success() public {
+    vm.expectEmit(true, false, false, true);
```

Comment:
What's the meaning of this? Maybe add inline comments to each param or one line above?

Reply:
For reference here are the docs:
https://book.getfoundry.sh/cheatcodes/expect-emit

I'll set the args to variable names to make it more clear.